design/accepted/kv_migration.md

```diff
@@ -0,0 +1,35 @@
+# KV Migration (export import phase 1) Design
+
+## Definitions
+1. KV Migration - The transition from schema based postgres implementation to key value store based postgres implementation
+2. DB Migration - The process of migrating postgres data using the postgres versioning schema
+3. KV Migration Version - “Schema” version for which the KV Migration will occur
+
+## Abstract
+Current migration mechanism relies on postgres versioning schema.
+KV Migration flow is a transitory mechanism, which will be needed only when upgrading lakeFS to the KV Migration Version. For efficiency and re-use we would like to create a preliminary version of the export-import feature to implement the KV Migration flow.
+
+## Considerations
+Although we cannot use the DB Migration flow to handle the KV migration, it must be incorporated into it and executed in the context of the DB migration
+DB migration must be performed up to the latest version before the KV Migration Version and only then KV Migration can happen
+KV Migration Version will be a consecutive version succeeding the last DB Migration version.
+
+## Migration Mechanism
+1. Migrate up command will perform schema migration up to the latest version
+2. Check if current version equals the KV Migration Version and perform KV Migration
+3. KV Migration will consist of 2 main steps: Export, Import
+4. Export:
+   1. Each migrating package will read the data from postgres into its corresponding KV models 
+   2. Data will be saved on file per pkg in it’s KV format 
+   3. Each pkg file will consist of a header containing the migration version and etc… 
+   4. Files will be saved in a designated export folder 
+5. Import:
+   1. Read files content from the ‘export’ folder directly to DB
+      
```

Comment:
How would you handle schema evolution of KV packages?

Reply:
Currently out of scope for this document - but once we are post KV migration, the migrate up command will be used to handle any schema changes as before, according to the current versioning scheme defined in the database. We do need to consider how do we want to save the version in the database and define the mechanism for the schema modifications.